conn.go

```diff
@@ -775,13 +774,22 @@ func (c *Conn) ReadPartitions(topics ...string) (partitions []Partition, err err
 // This method is exposed to satisfy the net.Conn interface but is less efficient
 // than the more general purpose WriteMessages method.
 func (c *Conn) Write(b []byte) (int, error) {
-	return c.WriteMessages(Message{Value: b})
+	return c.WriteMessagesCompressed(nil, Message{Value: b})
 }
 
 // WriteMessages writes a batch of messages to the connection's topic and
 // partition, returning the number of bytes written. The write is an atomic
 // operation, it either fully succeeds or fails.
 func (c *Conn) WriteMessages(msgs ...Message) (int, error) {
+	return c.WriteMessagesCompressed(nil, msgs...)
+}
+
+// WriteMessages writes a batch of messages to the connection's topic and
+// partition, returning the number of bytes written. The write is an atomic
+// operation, it either fully succeeds or fails.
+//
+// If the compression codec is not nil, the messages will be compressed.
+func (c *Conn) WriteMessagesCompressed(codec CompressionCodec, msgs ...Message) (int, error) {
```

Comment:
How about `WriteCompressedMessages`? So it reads like normal english :)

I don't have a strong preference, I like the common prefix too...

Reply:
That sounds good...I like normal English now that I think about it a little more.  The `Write` prefix should be plenty to indicate the functions are logically grouped.